apps/ledger-live-mobile/src/families/ethereum/EditTransaction/MethodSelection.tsx

```diff
@@ -0,0 +1,226 @@
+import { TrackScreen } from "../../../analytics";
+import { EditTransactionParamList } from "../../../components/RootNavigator/types/EditTransactionNavigator";
+import { StackNavigatorProps } from "../../../components/RootNavigator/types/helpers";
+import LText from "../../../components/LText";
+import { urls } from "../../../config/urls";
+
+type Props = StackNavigatorProps<
+  EditTransactionParamList,
+  ScreenName.EditTransactionMethodSelection
+>;
+
+export function MethodSelection({ navigation, route }: Props) {
+  const { operation, account, parentAccount } = route.params;
+
+  const transactionToEdit = fromTransactionRaw(
+    operation.transactionRaw! as TransactionRaw,
+  ) as Transaction;
+
+  const { transaction, setTransaction, status } =
+    useBridgeTransaction<Transaction>(() => {
+      return {
+        account,
+        parentAccount: parentAccount as Account,
+        transaction: transactionToEdit,
+      };
+    });
+
+  invariant(
+    transaction,
+    "[useBridgeTransaction - MethodSelection] could not found transaction from bridge.",
+  );
+
+  const mainAccount = getMainAccount(account, parentAccount as Account);
+  const feePerGas = new BigNumber(
+    EIP1559ShouldBeUsed(mainAccount.currency)
+      ? transactionToEdit.maxFeePerGas!
+      : transactionToEdit.gasPrice!,
+  );
+
+  const feeValue = new BigNumber(
+    transactionToEdit.userGasLimit! || transactionToEdit.estimatedGasLimit,
+  )
+    .times(feePerGas)
+    .div(new BigNumber(10).pow(mainAccount.unit.magnitude));
+
+  const haveFundToCancel = mainAccount.balance.gt(feeValue.times(1.3));
+  const haveFundToSpeedup = mainAccount.balance.gt(
+    feeValue
+      .times(1.1)
+      .plus(account.type === "Account" ? transactionToEdit.amount : 0),
+  );
+
+  let isOldestEditableOperation = true;
+
+  account.pendingOperations.forEach(pendingOperation => {
+    if (isEditableOperation(account, pendingOperation)) {
+      if (
+        operation.transactionSequenceNumber &&
+        pendingOperation.transactionSequenceNumber &&
+        pendingOperation.transactionSequenceNumber <
+          operation.transactionSequenceNumber
+      ) {
+        isOldestEditableOperation = false;
+      }
+    }
+  });
+
+  const currency = getAccountCurrency(account);
+
+  const bridge = getAccountBridge(account, parentAccount as Account);
+
+  const onSelect = (option: "cancel" | "speedup") => {
+    switch (option) {
+      case "cancel":
+        transactionToEdit.amount = new BigNumber(0);
+        transactionToEdit.data = Buffer.from("");
+        transactionToEdit.nonce = operation.transactionSequenceNumber;
+        transactionToEdit.allowZeroAmount = true;
+        transaction.mode = "send";
+
+        if (EIP1559ShouldBeUsed(currency)) {
+          if (transactionToEdit.maxPriorityFeePerGas) {
+            transactionToEdit.maxPriorityFeePerGas = new BigNumber(
+              transactionToEdit.maxPriorityFeePerGas.toNumber() * 1.1,
```

Comment:
This kind of multiplication will not work.
Transforming a BigNumber into a Number is risky and should never be done, because if that BigNumber is bigger than the max Integer value of Javascript, well you lost some precision along the way.

You can simply do something like this:
```typescript
const maxPriorityFeePerGas = transactionToEdit.maxPriorityFeePerGas.times(1.1);
```